changedetectionio/__init__.py

```diff
@@ -182,13 +182,23 @@ def hide_referrer(response):
 
     s_type = socket.AF_INET6 if ipv6_enabled else socket.AF_INET
 
-    if ssl_mode:
-        # @todo finalise SSL config, but this should get you in the right direction if you need it.
-        eventlet.wsgi.server(eventlet.wrap_ssl(eventlet.listen((host, port), s_type),
-                                               certfile='cert.pem',
-                                               keyfile='privkey.pem',
-                                               server_side=True), app)
-
-    else:
-        eventlet.wsgi.server(eventlet.listen((host, int(port)), s_type), app)
+    try:
+        if ssl_mode:
+            # @todo finalise SSL config, but this should get you in the right direction if you need it.
+            eventlet.wsgi.server(eventlet.wrap_ssl(eventlet.listen((host, port), s_type),
+                                                   certfile='cert.pem',
+                                                   keyfile='privkey.pem',
+                                                   server_side=True), app)
 
+        else:
+            eventlet.wsgi.server(eventlet.listen((host, int(port)), s_type), app)
+
+    except Exception as e:
+        import threading
+        import time
+        app.config.exit.set()
+        datastore.stop_thread = True
+        while len(threading.enumerate()) > 1:
+            time.sleep(1)
```

Comment:
```
#app.config.exit.set()
#datastore.stop_thread = True
while len(threading.enumerate()) > 1:
    print(threading.enumerate())
```
shows the reason.

Reply:
Also, this looks weird but refer to another apps's docker-compse, they start the DB first(https://github.com/Leantime/docker-leantime/blob/c35f5a72670d59efa7b2c0e88fca19ff4c11d8ae/docker-compose.yml#L30, https://github.com/minhng92/odoo-16-docker-compose/blob/c28f67b805413b7ebe7c8c4a91c1db38b67e2f83/docker-compose.yml#L18) and then the app. 

Because the CD is standalone,  the DB part started before the `wsgi`... So... then port conflict happens we need to stop the threads..